exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/CollectorIntegrationTest.java

```diff
@@ -194,6 +216,32 @@ void endToEnd() {
             stringKeyValue("otel_scope_version", "1.0.0"));
   }
 
+  private static void verifyHistogramMetric(ScopeMetrics testScopeMetrics) {
+    Optional<Metric> optRequestTotal =
+        testScopeMetrics.getMetricsList().stream()
+            .filter(metric -> metric.getName().equals("latency"))
+            .findFirst();
+    assertThat(optRequestTotal).isPresent();
+    Metric requestTotal = optRequestTotal.get();
+    assertThat(requestTotal.getDataCase()).isEqualTo(Metric.DataCase.HISTOGRAM);
+
+    Histogram requestHistogram = requestTotal.getHistogram();
```

Comment:
This demonstrates that the configuration is not actually working. If this was an exponential histogram, `requestTotal.hasExponentialHistogram()` would be true, but its not and you're getting explicit bucket histograms instead.

In order to scrape exponential histograms, a client needs to set the accept header to `Accept: application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily`. This triggers the prometheus server to return the binary representation which includes the exponential histogram representation. This test is using the otel collector to scrape the prometheus metrics from the opentelemetry java SDK's prometheus exporter. The collector can be configured to request the binary representation when scraping as [described here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#prometheus-native-histograms) (although I haven't tried it myself).

Reply:
I think it works as a native histogram in `CollectorIntegrationTest.java` since the number of buckets it creates for these data points is 1. At the same time, if this would have been an Explicit Bucket (which will happen if we remove `DefaultAggregationSelector=base2ExponentialBucketHistogram` from `PrometheusHttpServer`), then it will give 16 buckets for the same data points.

So this confirms that even for histograms which are not exponential, with `DefaultAggregationSelector` being set to `base2ExponentialBucketHistogram` , aggregation is indeed Native (or base2Exponential).

Reply:
This seems like a pretty brittle signal that exponential histograms have in fact been configured. Why don't we take a simpler approach and add a test to `PrometheusMetricReaderTest` which directly calls `PrometheusMetricReader#collect()`, and assert against the response `MetricSnapshots`?

Reply:
Pushed a commit to resolve this.